### PR TITLE
Editorial: refactor if statement in call expression

### PR DESCRIPTION
**Pull request by devsnek**
Just a small change that gets rid of a superfluous if statement

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -13804,14 +13804,13 @@ <h1>Runtime Semantics: Evaluation</h1>
...
           1. Let _func_ be ? GetValue(_ref_).
-          1. If Type(_ref_) is Reference, IsPropertyReference(_ref_) is *false*, and GetReferencedName(_ref_) is *"eval"*, then
-            1. If SameValue(_func_, %eval%) is *true*, then
-              1. Let _argList_ be ? ArgumentListEvaluation of _arguments_.
-              1. If _argList_ has no elements, return *undefined*.
-              1. Let _evalArg_ be the first element of _argList_.
-              1. If the source code matching this |CallExpression| is strict mode code, let _strictCaller_ be *true*. Otherwise let _strictCaller_ be *false*.
-              1. Let _evalRealm_ be the current Realm Record.
-              1. [id="step-callexpression-evaluation-direct-eval"] Return ? PerformEval(_evalArg_, _evalRealm_, _strictCaller_, *true*).
+          1. If Type(_ref_) is Reference, IsPropertyReference(_ref_) is *false*, and GetReferencedName(_ref_) is *"eval"*, and SameValue(_func_, %eval%) is *true*, then
```
Comment (jmdyck) [inline spec.html:13807]:
```suggestion
          1. If Type(_ref_) is Reference, IsPropertyReference(_ref_) is *false*, GetReferencedName(_ref_) is *"eval"*, and SameValue(_func_, %eval%) is *true*, then
```

**Comment by leobalter** [comment]
Inlining too many things makes it really hard for me to read and keep track of all the conditions. It's always ridiculously easier for me to read each condition line by line so I don't think the nested if is superfluous. 

I'd hope we can reconsider this change and avoid inlining everything.

**Comment by devsnek** [comment]
it would make sense to investigate ways to make long lines more readable, but in this case it was confusing for me to see the separate if with no else, I think this communicates the intent better.

**Comment by leobalter** [comment]
Unfortunately the reasons for me to have an easier time reading shorter conditional lines is personal and I don't intend to demonstrate how or anything.

I hope we consider different biases for what communicates better.

**Comment by rkirsling** [comment]
The issue is that nesting a conditional here leads the reader to expect that it's been done meaningfully.
For readability, we could either give names to the conjuncts or add newlines, just like with surface code.

**Comment by jmdyck** [comment]
If you abstracted away the whole condition, you could almost call it `IsDirectEval`. Technically, it's only a direct eval if control gets to 6.a.vi, so if the condition succeeds, but ArgumentListEvaluation throws, or there aren't any args, then it isn't a direct eval. Mind you, I don't think there'd be any normative difference in saying that anything that passes the condition is a direct eval.

**Comment by claudepache** [comment]
In this particular case, having the two conditions split in two lines is clearer to me, because:
* the first one (that checks basically whether the expression looks like a direct eval) can be checked statically—and I guess that engines do check it statically in order to see if they are allowed to perform certain optimisations;
* the second one (that checks that the `eval` binding contains `%eval%`) can only be checked at runtime.

The intent would probably be clearer if the first condition were abstracted out in an `IsDirectEvalSyntax` abstract operation (either as static semantics, or with a note stating that it could be checked statically). Then, the two `if` statements could be merged into one if you want:

>  If IsDirectEvalSyntax of _CallExpression_ is **true** and  SameValue(_func_, %eval%) is ***true***, then

**Comment by syg** [comment]
We talked about this on the editor call. We're all pretty neutral on this current PR, but find @claudepache's suggestion in https://github.com/tc39/ecma262/pull/2105#issuecomment-661881682 compelling as an editorial improvement. Does anyone want to take a shot at writing that PR?

**Pull request closed without merging**